packages/firebase_auth/firebase_auth_web/lib/src/utils/web_utils.dart

```diff
@@ -4,38 +4,24 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:io';
+import 'dart:js_interop';
 
 import 'package:firebase_auth_platform_interface/firebase_auth_platform_interface.dart';
 import 'package:firebase_auth_web/firebase_auth_web.dart';
 import 'package:firebase_auth_web/src/firebase_auth_web_multi_factor.dart';
 import 'package:firebase_core_web/firebase_core_web_interop.dart'
     as core_interop;
-import 'package:js/js_util.dart';
+import 'package:firebase_core_web/firebase_core_web_interop.dart';
 
 import '../interop/auth.dart' as auth_interop;
 import '../interop/multi_factor.dart' as multi_factor_interop;
-import '../interop/window_interop.dart' as window_interop;
 
-/// Workaround test to check whether `e` is practically a FirebaseError.
-///
-/// Ideally we'd check whether `e instanceof FirebaseError` however
-/// there are two definitions of `FirebaseError` in deployed apps, one from
-/// `firebase-auth.js` (which is usually minified) and one from
-/// `firebase-app.js`.
-///
-/// Because they are not the same, the instanceof check fails. Instead, we test
-/// that it is a window.Error (since that's the base class of FirebaseError) and
-/// we check that it defines "customData" property which is on the AuthError but not the FirebaseError it extends
-bool _isFirebaseAuthError(Object e) =>
-    instanceof(e, window_interop.errorConstructor) &&
-    hasProperty(e, 'customData');
-
-bool _hasFirebaseAuthErrorCodeAndMessage(Object e) {
-  if (_isFirebaseAuthError(e)) {
-    String? code = getProperty(e, 'code');
-    String? message = getProperty(e, 'message');
-    if (code == null || !code.startsWith('auth/')) return false;
-    if (message == null || !message.contains('Firebase')) return false;
+bool _hasFirebaseAuthErrorCodeAndMessage(JSError e) {
+  if (e.name == 'FirebaseError') {
+    String code = e.code ?? '';
+    String message = e.message ?? '';
+    if (code.startsWith('auth/')) return false;
```

Comment:
Are these checks correct? In the previous version of this, the checks were
```dart
    if (code == null || !code.startsWith('auth/')) return false;
    if (message == null || !message.contains('Firebase')) return false;
 ```

Now it is inverted to return false if the message contains `Firebase`, which seems wrong?